nats/tests/consumer_command_test.go

```diff
@@ -0,0 +1,379 @@
+		if !re.MatchString(string(output)) {
+			t.Errorf("expected match for field %q with pattern %q in output:\n%s", field, re.String(), output)
+		}
+	}
+}
+
+func TestConsumerCopy(t *testing.T) {
+	srv, name, cleanup := setupConsumerTest(t)
+	defer cleanup()
+
+	output := runNatsCli(t, fmt.Sprintf("--server='%s' consumer copy ORDERS %s COPY_1", srv.ClientURL(), name))
+	fields := map[string]*regexp.Regexp{
+		"header":             regexp.MustCompile(`Information for Consumer ORDERS > COPY_1 created \d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}[+-]\d{2}:\d{2}`),
+		"name":               regexp.MustCompile(`Name:\s+COPY_1`),
+		"pull_mode":          regexp.MustCompile(`Pull Mode:\s+true`),
+		"deliver_policy":     regexp.MustCompile(`Deliver Policy:\s+All`),
+		"ack_policy":         regexp.MustCompile(`Ack Policy:\s+Explicit`),
+		"ack_wait":           regexp.MustCompile(`Ack Wait:\s+30\.00s`),
+		"replay_policy":      regexp.MustCompile(`Replay Policy:\s+Instant`),
+		"max_ack_pending":    regexp.MustCompile(`Max Ack Pending:\s+1,000`),
+		"max_waiting_pulls":  regexp.MustCompile(`Max Waiting Pulls:\s+512`),
+		"host_version":       regexp.MustCompile(`Host Version:\s+\d+\.\d+\.\d+(-[0-9A-Za-z.-]+)?(\+[0-9A-Za-z.-]+)?`),
+		"required_api_level": regexp.MustCompile(`Required API Level:\s+\d+ hosted at level \d+`),
+		"last_delivered_msg": regexp.MustCompile(`Last Delivered Message:\s+Consumer sequence: 0 Stream sequence: 0`),
+		"ack_floor":          regexp.MustCompile(`Acknowledgment Floor:\s+Consumer sequence: 0 Stream sequence: 0`),
+		"outstanding_acks":   regexp.MustCompile(`Outstanding Acks:\s+0 out of maximum 1,000`),
+		"redelivered_msgs":   regexp.MustCompile(`Redelivered Messages:\s+0`),
+		"unprocessed_msgs":   regexp.MustCompile(`Unprocessed Messages:\s+0`),
+		"waiting_pulls":      regexp.MustCompile(`Waiting Pulls:\s+0 of maximum 512`),
+	}
+
+	for field, re := range fields {
+		if !re.MatchString(string(output)) {
+			t.Errorf("expected match for field %q with pattern %q in output:\n%s", field, re.String(), output)
+		}
+	}
+}
+
+func TestConsumerNext(t *testing.T) {
+	srv, name, cleanup := setupConsumerTest(t)
+	defer cleanup()
+	msgBody := "test message"
+
+	runNatsCli(t, fmt.Sprintf("--server='%s' publish ORDERS.foo --count=1 \"%s\"", srv.ClientURL(), msgBody))
+	output := runNatsCli(t, fmt.Sprintf("--server='%s' consumer next ORDERS %s", srv.ClientURL(), name))
+
+	if !regexp.MustCompile(msgBody).MatchString(string(output)) {
+		t.Errorf("failed to find message body %s in %s", msgBody, string(output))
+	}
+}
+
+func TestConsumerSub(t *testing.T) {
+	srv, name, cleanup := setupConsumerTest(t)
+	defer cleanup()
+	msgBody := "test message"
+
+	runNatsCli(t, fmt.Sprintf("--server='%s' publish ORDERS.foo --count=1 \"%s\"", srv.ClientURL(), msgBody))
+	output := runNatsCli(t, fmt.Sprintf("--server='%s' consumer sub ORDERS %s", srv.ClientURL(), name))
+
+	if !regexp.MustCompile(msgBody).MatchString(string(output)) {
```

Comment:
we do a lot of this, could probably make a helper `expectMatchRegex(t, string(output), msgBody)` or similar?